FoodplannerDataAccessSql/Codes/OneTimePasswordRepository.cs

```diff
@@ -0,0 +1,129 @@
+
+            // We assume if none found it is an expired code
+            if (result == null)
+                return true;
+
+            return result <= DateTime.UtcNow;
+        }
+
+        public async Task<int> DeleteAsync(string code)
+        {
+            const string sql = "DELETE FROM one_time_password WHERE code = @Code";
+
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+
+            var rowsAffected = await connection.ExecuteAsync(sql, new { Code = code });
+
+            return rowsAffected;
+        }
+
+        public async Task<OneTimePassword?> GetFromCodeAsync(string code)
+        {
+            const string sql = "SELECT * FROM one_time_password WHERE code = @Code";
+
+            using var connection = _connectionFactory.Create();
+            connection.Open();
+            var result = await connection.QuerySingleOrDefaultAsync<OneTimePassword>(sql, new { Code = code });
+            return result;
+        }
+
+
+        public async Task<int> InsertAsync(OneTimePassword createOTP)
+        {
+            var sql = "INSERT INTO one_time_password (generated_by, code, created_on, expires_on, used, used_by_user, child_user) VALUES (@GeneratedBy, @Code, @CreatedOn, @ExpiresOn, @Used, @UsedByUser, @ChildUser) RETURNING code_id";
+
+            using (var connection = _connectionFactory.Create())
+            {
+                connection.Open();
+                var result = await connection.QuerySingleAsync<int>(sql, new
+                {
+                    GeneratedBy = createOTP.GeneratedBy,
+                    Code = createOTP.Code,
+                    CreatedOn = createOTP.CreatedOn,
+                    ExpiresOn = createOTP.ExpiresOn,
+                    Used = createOTP.Used,
+                    UsedByUser = createOTP.UsedByUser,
+                    ChildUser = createOTP.ChildUser
+                });
+                return result;
+            }
+        }
+
+        public async Task<int> UpdateAsync(OneTimePassword OTP)
+        {
+            var sql = @"
+    UPDATE one_time_password 
+    SET 
+        generated_by = @GeneratedBy,
+        code = @Code,
+        created_on = @CreatedOn,
+        expires_on = @ExpiresOn,
+        used = @Used,
+        used_by_user = @UsedByUser,
+        child_user = @ChildUser
+    WHERE code_id = @CodeId;
+";
+
+            using (var connection = _connectionFactory.Create())
+            {
+                connection.Open();
+                var result = await connection.ExecuteAsync(sql, new
+                {
+                    CodeId = OTP.CodeId,
+                    GeneratedBy = OTP.GeneratedBy,
+                    Code = OTP.Code,
+                    CreatedOn = OTP.CreatedOn,
+                    ExpiresOn = OTP.ExpiresOn,
+                    Used = OTP.Used,
+                    UsedByUser = OTP.UsedByUser,
+                    ChildUser = OTP.ChildUser
+                });
```

Comment:
Remember to close the connection when you are done with it:
```suggestion
                });
                connection.Close();
```